flink-runtime/src/test/java/org/apache/flink/runtime/source/coordinator/SourceCoordinatorAlignmentTest.java

```diff
@@ -131,6 +136,73 @@ void testWatermarkAlignmentWithTwoGroups() throws Exception {
         }
     }
 
+    /**
+     * When JobManager failover and auto recover job, SourceCoordinator will reset twice: 1. Create
+     * JobMaster --> Create Scheduler --> Create DefaultExecutionGraph --> Init
+     * SourceCoordinator(but will not start it) 2. JobMaster call
+     * restoreLatestCheckpointedStateInternal, which will call {@link
+     * org.apache.flink.runtime.operators.coordination.RecreateOnResetOperatorCoordinator#resetToCheckpoint(long,byte[])}
+     * and reset SourceCoordinator. Because the first SourceCoordinator is not be started, so the
+     * period task can't be stopped.
+     */
+    @Test
+    void testSourceCoordinatorReset() throws Exception {
```

Comment:
In general, this test is checking that the `announceCombinedWatermark` shouldn't be called without `SourceCoordinator#start`, the reset is just a case of this test(Currently there is only one case.).  right?

If so, I think the method name isn't general. How about `testAnnounceCombinedWatermarkWithoutStart`?

Reply:
yeah, done